### ISSUE TITLE
ProvisionedThroughputExceededException handling

### ISSUE THREAD

**Issue by glaphil** [scanamo/scanamo#164]
Hello, 

How to handle this exception with scanamo ?

I'm forced to wrap the call to scanamo into a Try, which force me to do 2 successive pattern matching
(one on the Try and the other on the Eithet)

`
 Try[Seq[Either[DynamoReadError, DynamoCustomerEvent]]] = Try(Scanamo.exec(client)(Table[DynamoCustomerEvent](eventsCustomerByDayTableName)
      .query('src_customer_id -> partitionKey(source,id) and ('event_date between Bounds(start_date, end_date)))))
`
I would have liked to get it in the Either return of scanamo.exec

`Seq[Either[DynamoReadError, DynamoCustomerEvent]]] = Try(Scanamo.exec(client)(Table[DynamoCustomerEvent](eventsCustomerByDayTableName)
      .query('src_customer_id -> partitionKey(source,id) and ('event_date between Bounds(start_date, end_date)))`

Is it possible ? Or could you add this feature to scanamo ?

Thanks.

Philippe

**Comment by philwills** [comment]
There's a general issue in that Scanamo doesn't encode errors during the interaction with Dynamo as values. This is very much something I'd like to support, but it's not a small change. The minimal change would be to add a new interpreter in the style of the ones here https://github.com/scanamo/scanamo/blob/master/scanamo/src/main/scala/com/gu/scanamo/ops/ScanamoInterpreters.scala.

**Comment by dmitry-avershin** [comment]
hi @philwills can I somehow help to implement this issue? We also faced it in our project as we relied that exceptions don't escape the either.

**Comment by philwills** [comment]
@dmitry-avershin yes, absolutely. As mentioned above, I think the simplest approach would be to write an alternative interpreter that catches the errors and contains them.

I've not explored it properly, as I'm no longer working with the library day to day, but I think the cats effect module has an interpreter which does that, so it might be possible to use that to achieve the desired effect.

**Comment by regiskuckaertz** [comment]
You may be able to get away with just adding a constraint `MonadError` on the effect `F`.

**Comment by dmitry-avershin** [comment]
@regiskuckaertz, you mean in my application code?

**Comment by regiskuckaertz** [comment]
In the cats/scalaz interpreters

**Comment by dmitry-avershin** [comment]
@regiskuckaertz I see, thx, however I couldn't find a way to use cats interpreters, they are not a part of the "scanamo" package and I couldn't find separate module for them in maven, are they released?

**Comment by regiskuckaertz** [comment]
Oh hang on, you're right, sorry about that. I think it's time for another milestone release, then it will be available as `scanamo-cats-effect`.

**Comment by dmitry-avershin** [comment]
yeah, that what I thought, thx, I would wait until its released and see whether I can extend it to collect errors or something
